prow/cmd/build/controller.go

```diff
@@ -685,6 +685,19 @@ func decorateSteps(steps []coreapi.Container, dc prowjobv1.DecorationConfig, too
 		if i > 0 {
 			previousMarker = entries[i-1].MarkerFile
 		}
+
+		if dc.Timeout == nil {
+			dc.Timeout = &prowjobv1.Duration{
+				Duration: 0,
+			}
+		}
+
+		if dc.GracePeriod == nil {
+			dc.GracePeriod = &prowjobv1.Duration{
+				Duration: 0,
+			}
+		}
```

Comment:
Why is this defaulting implemented in the build controller? These values should already have been defaulted before the build controller handles them.

Reply:
I was going to make this comment then I channelled my inner @fejta and there is technically nothing stopping someone from making a `nil` entry with `mkpj` or something and then the defensive coding here is ok?

Reply:
/shrug
We can keep it if that is a concern, but I'd like to see the proper defaulting as well at least (especially because there is the potential for segfaults with the current implementation).

`mkpj` will still properly default the field based on the default_decoration_config. The only concern would be hand crafted ProwJob yaml which can break prow in a lot of ways already. I think adding a ProwJob webhook admission controller is the right way to handle that.

Reply:
We could add a defaulting accessor to the config struct and disallow the use of direct references @droslean

Reply:
addressed